### PR TITLE
Ingest run_constraints on other llvm packages from repodata patches

### LINKED ISSUE #141: add clang, clang-tools, and other pins to recipe
Our repodata patches keep adding the following to new releases:

```
linux-64::llvm-13.0.1-h32600fe_0.tar.bz2
-    "llvmdev   13.0.1"
+    "llvmdev   13.0.1",
+    "clang 13.0.1.*",
+    "clang-tools 13.0.1.*",
+    "llvm-tools 13.0.1.*"
linux-64::llvm-tools-13.0.1-hf817b99_0.tar.bz2
-    "llvmdev   13.0.1"
+    "llvmdev   13.0.1",
+    "clang 13.0.1.*",
+    "clang-tools 13.0.1.*",
+    "llvm 13.0.1.*"
linux-64::llvmdev-13.0.1-hf817b99_0.tar.bz2
-  "version": "13.0.1"
+  "version": "13.0.1",
+  "constrains": [
+    "clang 13.0.1.*",
+    "clang-tools 13.0.1.*",
+    "llvm 13.0.1.*",
+    "llvm-tools 13.0.1.*"
+  ]
```

I propose we either timebox the patches or add those pins here. 

Thoughts?

xref: https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/issues/223

### PR DESCRIPTION
**Pull request by h-vetinari**
Closes #141 (hopefully)

Still a draft because I'm not sure about some things (will comment in the diff).

### CODE REVIEW

**Comment by h-vetinari** [comment]
Only https://github.com/conda-forge/llvmdev-feedstock/pull/142/commits/9217917b29f611fd5d78be59992a998dc9e554c7 is worth reviewing...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -56,6 +56,11 @@ outputs:
         - {{ pin_subpackage("libllvm" + major_ver, exact=True) }}
         - {{ pin_subpackage("llvm-tools", exact=True) }}
         - libcxx >={{ cxx_compiler_version }}.a0                        # [osx]
+      run_constrained:
+        - llvm        {{ version }}
+        - llvm-tools  {{ version }}
+        - clang       {{ version }}
+        - clang-tools {{ version }}
```
Comment (h-vetinari) [inline recipe/meta.yaml:63]: Adding `llvm-tools` as a run-constraint when it's already pinned as a subpackage seems... unnecessary?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -103,8 +108,11 @@ outputs:
         - zlib
       run:                                                            # [not win]
         - {{ pin_subpackage("libllvm" + major_ver, exact=True) }}     # [not win]
-      run_constrained:                                                # [not win]
-        - llvmdev   {{ version }}                                     # [not win]
```
Comment (h-vetinari) [inline recipe/meta.yaml]: This part I don't understand either. The run-constraint _from_ `llvm` _on_ `llvmdev` should have been there already, but it still shows up in the diff of #141.

**Comment by beckermr** [comment]
I'll need to resolve the merge conflicts at the cli. Will do later.

**Pull request merged**